nshutijonathan/Auto-mart3
# Ft2 database set up 166413022

#### What does this PR do?
V2  of Auto-Mart API should use Postgres SQL
Have SQL configured 
Have endpoints user endpoints built 
Have the endpoints tested 
#### Description of Task to be completed?
-Implemented TDD practice
-Created User endpoints
-Implemented Authorizations
-Protected Admin Routes
Have The following endpoint working and tested 
*   POST/localhost/API/v2/auth/signup
*   POST/localhost/API/v2/auth/signin
*   POST/localhost/API/v2/auth/signup/admin
*   GET/localhost/API/v2/users
*   GET/localhost:3000/api/v2/users/<:id>
*   DELETE/localhost:3000/api/v2/users/<:id>
#### How should this be manually tested?
* Download or clone the tutorial project code from [this repo](https://github.com/nshutijonathan/Auto-mart3.git), Install all required NPM  packages by running NPM install from the command line in the project root folder (where the package.json is located).
* Start the API by running NPM  start from the command line in the project root folder, you should see the message Server listening on port 3000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
* to test authentication and authorization, add this to the header 
Key: :x-auth-token,value:JWT TOKEN
#### Any background context you want to provide?
* There is Implementation of Javascript(ES6) logic with data structures to persist some data
* Used mocha and chai to test all the endpoints 
#### What are the relevant pivotal tracker stories?
[#166413022](https://www.pivotaltracker.com/n/projects/2346053/stories/166413022)

#### Screenshots
![database_configured](https://user-images.githubusercontent.com/16412410/59107739-90faaf80-8939-11e9-9f20-d65780d6a0f8.PNG)

--
